src/repository/room.repository.ts

```diff
@@ -71,6 +72,34 @@ export const findRooms = async (pageNum: number): Promise<RoomOverview[]> => {
   });
   return roomOverviews;
 };
+
+export const findRecentRooms = async (userId: string) => {
+  var roomOverviews: RoomOverview[] = [];
+  const rooms = await client.study_history.findMany({
+    where: { user_id: userId, room: { deleted_at: null } },
+    distinct: "room_id",
+    take: MAX_RECENT_ROOM_NUM,
+    orderBy: { join_at: "desc" },
+    include: {
+      room: { include: { study_history: { where: { exit_at: null } } } },
+    },
+  });
+  await rooms.map(async (rooms) => {
+    const roomOverview = new RoomOverview(
+      rooms.room.id,
+      rooms.room.master_id,
+      rooms.room.title,
+      rooms.room.password ? true : false,
+      rooms.room.thumbnail,
+      rooms.room.study_history.length,
+      MAX_ROOM_CAPACITY,
+      [] //room.room_tags
+    );
+    roomOverviews.push(roomOverview);
+  });
+  return roomOverviews;
```

Comment:
roomOverviews라는 변수를 만들지 않고도 결과를 반환할 수 있습니다. 그리고 map 함수는 async 함수가 아니기 때문에 await 키워드는 없어야합니다.(https://velog.io/@minsangk/2019-09-06-0209-작성됨-eik06xy8mm)

```suggestion
  return rooms.map((rooms) => {
     return new RoomOverview(
       rooms.room.id,
       rooms.room.master_id,
       rooms.room.title,
       rooms.room.password ? true : false,
       rooms.room.thumbnail,
       rooms.room.study_history.length,
       MAX_ROOM_CAPACITY,
       [] //room.room_tags
     );
   });
```

Reply:
그렇네요 ! 감사합니다. 위에 getRoom도 수정했습니다.